openshift-kube-apiserver/admission/scheduler/nodeenv/admission.go

```diff
@@ -62,6 +62,11 @@ func (p *podNodeEnvironment) admit(ctx context.Context, a admission.Attributes,
 		return nil
 	}
 
+	// NodeName has a precedence
+	if len(pod.Spec.NodeName) > 0 {
```

Comment:
@atiratree are you sure we want to allow setting nodename explicitly on a pod? Isn't the goal of project node selectors to ensure we can restrict authors to running only on those explicitly pointed out nodes?

Reply:
Yes you are right, scheduling should not take a precedence over project node selector restrictions.

I have updated the PR, to increase the validations.